src/bundle/Resources/config/trusted_device.php

```diff
@@ -15,9 +18,22 @@
 
 return static function (ContainerConfigurator $container): void {
     $container->services()
-        ->set('scheb_two_factor.trusted_jwt_encoder', JwtTokenEncoder::class)
+        ->set('scheb_two_factor.trusted_jwt_encoder.configuration.algorithm', Sha256::class)
+
+        ->set('scheb_two_factor.trusted_jwt_encoder.configuration.key', InMemory::class)
+            ->factory([InMemory::class, 'plainText'])
             ->args(['%kernel.secret%'])
```

Comment:
I'd add an option to the bundle's configuration (defaulting to `%kernel.secret%`) to set the key for the trusted device token.

That would give people the ability to easily change the key for the trusted device token only, without changing the kernel.secret for the entire application and without overwrite the a service definition from the bundle (as you have demonstrated in [app/config/services.yaml](https://github.com/scheb/2fa/pull/158/files#diff-763476bab89e87b6546d922380ad2a190a43254a2ff04d178c55ab92b90ba813)). Overwriting services from 3rd party bundles just be the last resort.

Reply:
Adding as configuration makes sense, the only thing is that we'd need to be able to distinguish b64 encoded chars and plain strings.

I'm saying that because `random_bytes(32)` is usually used to generate keys, then it needs to be b64 encoded to be used in config files.

Reply:
I wouldn't overcomplicate things here, I'd make it behave the same as `kernel.secret` => plain text value with minimum length of 32 characters (265 bit). People are obviously capable of configuring the `kernel.secret` value, so I'd expect them to be able doing the same for an (optional) JWT token value in the bundle configuration :D

Reply:
I don't think it's overcomplicated... Based on [this](https://github.com/symfony/flex/blob/v2.2.3/src/Configurator/EnvConfigurator.php#L202) `kernel.secret` is a hexadecimal representation of random bytes...

Up to you, though 👍

Reply:
I believe that's just their cheap way of making random bytes YAML-safe, when those config files are generated via Symfony Flex ;) I don't think that value is actually _required_ to be hexadecimal and decoded `hex2bin` in Symfony.

https://github.com/symfony/symfony/search?q=hex2bin Doesn't look like the secret being decoded, likely just used as raw bytes.

So anyways, I'd be totally fine with making that config option in the bundle also raw bytes.